CoViewingExample/ExampleAVPlayerSource.swift

```diff
@@ -0,0 +1,234 @@
+    static private var useDisplayLinkTimer = true
+
+    init(item: AVPlayerItem) {
+        sampleQueue = DispatchQueue(label: "com.twilio.avplayersource", qos: DispatchQoS.userInteractive,
+                                    attributes: DispatchQueue.Attributes(rawValue: 0),
+                                    autoreleaseFrequency: DispatchQueue.AutoreleaseFrequency.workItem,
+                                    target: nil)
+        super.init()
+
+        let presentationSize = item.presentationSize
+        let presentationPixels = presentationSize.width * presentationSize.height
+        print("Prepare for player item with size:", presentationSize, " pixels:", presentationPixels);
+
+        /*
+         * We might request buffers downscaled for streaming. The output will be NV12, and backed by an IOSurface
+         * even though we dont explicitly include kCVPixelBufferIOSurfacePropertiesKey.
+         */
+        let attributes: [String : Any]
+
+        // TODO: We need to interrogate the content and choose our range (video/full) appropriately.
+        if (presentationSize.width > ExampleAVPlayerSource.kFrameOutputMaxDimension ||
+            presentationSize.height > ExampleAVPlayerSource.kFrameOutputMaxDimension) {
+            let streamingRect = AVMakeRect(aspectRatio: presentationSize, insideRect: ExampleAVPlayerSource.kFrameOutputMaxRect)
+            print("Requesting downscaling to:", streamingRect.size, ".");
+
+            attributes = [
+                kCVPixelBufferWidthKey as String : Int(streamingRect.width),
+                kCVPixelBufferHeightKey as String : Int(streamingRect.height),
+                kCVPixelBufferPixelFormatTypeKey as String : kCVPixelFormatType_420YpCbCr8BiPlanarFullRange
+                ] as [String : Any]
+        } else {
+            attributes = [
+                kCVPixelBufferPixelFormatTypeKey as String : kCVPixelFormatType_420YpCbCr8BiPlanarFullRange
+                ] as [String : Any]
+        }
+
+        videoOutput = AVPlayerItemVideoOutput(pixelBufferAttributes: attributes)
+        videoOutput?.setDelegate(self, queue: sampleQueue)
+
+        if ExampleAVPlayerSource.useDisplayLinkTimer {
+            addDisplayTimer()
+        }
+        videoOutput?.requestNotificationOfMediaDataChange(withAdvanceInterval: 0.02)
+
+        item.add(videoOutput!)
+    }
+
+    func outputFrame(itemTimestamp: CMTime) {
+        guard let output = videoOutput else {
+            return
+        }
+        guard let consumer = captureConsumer else {
+            return
+        }
+        if !output.hasNewPixelBuffer(forItemTime: itemTimestamp) {
+            // TODO: Consider suspending the timer and requesting a notification when media becomes available.
+//            print("No frame for host timestamp:", CACurrentMediaTime(), "\n",
+//                  "Last presentation timestamp was:", lastPresentationTimestamp != nil ? lastPresentationTimestamp! : CMTime.zero)
```

Comment:
fyi - this file has commented logs

Reply:
In this case, we should log once, suspend the timer and restart it later. This should be an easy fix. 👍